client/lib/domains/dns/reducer.js

```diff
@@ -46,6 +44,36 @@ function deleteDns( state, domainName, record ) {
 	return update( state, command );
 }
 
+function markDnsForDeletion( state, domainName, record, isBeingDeleted ) {
+	const index = findDns( state, domainName, record ),
+		updatedRecord = Object.assign( {}, record, {
+			isBeingDeleted,
+		} );
+
+	if ( index === -1 ) {
+		return state;
+	}
+
+	const command = {
+		[ domainName ]: {
+			records: {
+				[ index ]: {
+					$merge: updatedRecord
+				}
+			}
+		}
+	};
+
+	return update( state, command );
+}
+
+function findDns( state, domainName, record ) {
```

Comment:
`findDns` is probably not the right name for a function that returns index. How about `findDnsIndex`?